app/seeders/oauth_applications_seeder.rb

```diff
@@ -0,0 +1,59 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+class OAuthApplicationsSeeder < Seeder
+  def seed_data!
+    call = create_app
+    unless call.success?
+      print_error "Seeding mobile oauth application failed:"
+      call.errors.full_messages.each do |msg|
+        print_error "  #{msg}"
+      end
+    end
+  end
+
+  def applicable?
+    Doorkeeper::Application.where(builtin: true).empty?
+  end
+
+  def not_applicable_message
+    "No need to seed oauth appplications as they are already present."
+  end
+
+  def create_app
+    OAuth::Applications::CreateService
+      .new(user: User.system)
+      .call(
+        enabled: true,
+        name: "OpenProject Mobile App",
+        redirect_uri: "openprojectapp://oauth-callback",
```

Comment:
I guess that this OAuth application is a generic application. It's usage is not limited to a certain mobile app. I am thinking of other user apps, such as a CLI or Solibri integration. They would need exactly the same stuff but are no mobile app. That is why I propose a more generic name and callback URL protocol

```suggestion
        name: "OpenProject client (e.g mobile app)"
        redirect_uri: "openproject://oauth-callback",
```

Reply:
The application cannot truly be generic unless you provide the correct redirect URI. We might guess some common ones such as urn:ietf:wg:oauth:2.0:oob but that will only get us so far. Why not add multiple builtin apps then? This would allow users to enable/disable them selectively instead of re-using the one app

Reply:
Another idea would be to seed a couple of allowed redirect URLs (an oauth application can have multiple). However, then we would ideally make the seeded ones not delectable... complex. However, we could allow adding more redirect URLs in case someone needs a different one.

No strong opinion here.

Reply:
+1 to that.

Reply:
From what I heard in the last discussions, we will start with an mobile specific oauth app. When new built-in apps are planned, we either can rename this or create more built-in apps with specific names.